### PR TITLE
Refactor install procedure to support automated setup

### PR DESCRIPTION
**Pull request by audebert**
* The new install process has only *one* clone of sadm. Every instances of sadm are copies of this first clone. This means that you can clone once from, use a different branch or make a modification and *every* instance of sadm will have it. If you work with container, the current folder will be used as the sources of all the sadm in the test setup.
* Reworked install procedure to be more modular. The new process follows these steps:

1. Hardware setup. Work with disks, RAID1, boot process
2. Arch Linux setup. Common to all Arch Linux installs
3. SADM setup. Common SADM configuration
4. System specific setup (gw, rhfs, web, users)

* If the docs and the container scripts disagree, the container script should probably used as ground truth.
* When in doubt, follow the container script.
* Check out the brand new checks/ folder for awesome (yet still a very limited) system sanity checks.
* Reworked the network setup configuration to be less braindead.

### CODE REVIEW

**Comment by Zopieux** [comment]
Ugh I need an actual working setup to test & review that. Will not have the time until the finals. :slightly_frowning_face:

File: etc/systemd/system/conntrack.service [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=Conntrack logging of new connections, in case of complaint
+After=network-online.target
+
+[Service]
+ExecStart=/usr/bin/conntrack -E -e NEW
```
Comment (Zopieux) [inline etc/systemd/system/conntrack.service]: No `Restart=always`?

File: install.py [hunk 2]
```diff
@@ -607,7 +624,8 @@ def install_udbsync_rootssh():
 
 def install_presenced():
     requires('libprologin')
-    requires('nginxcfg')
+    # FIXME(halfr): remove if proven not useful
```
Comment (Zopieux) [inline install.py]: Needed because `install_nginx_service`?

File: install_scripts/bootstrap_arch_linux.sh [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+# Usage
+if [ $# -ne 3 ]; then
+    echo >&2 "Usage: $0 ROOT_DIR HOSTNAME PLAINTEXT_ROOT_PASSWORD_FILE"
+    echo >&2 ""
+    echo >&2 "Install and configure Arch Linux in the ROOT_DIR folder."
+    echo >&2 "Example (as root):"
+    echo >&2 "  $ echo my_root_password_is_pretty_long > plaintext_root_pass"
+    echo >&2 "  $ mkdir gw"
+    echo >&2 "  $ ./bootstrap_arch_linux.sh ./gw gw ./plaintest_root_pass¬"
```
Comment (Zopieux) [inline install_scripts/bootstrap_arch_linux.sh]: What's that `¬`?

File: install_scripts/bootstrap_arch_linux.sh [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+hostname="$2"
+root_password_file="$3"
+
+if [ ! -r $root_password_file ]; then
+    echo >&2 "Error: password file '$root_password_file' must be readable"
+    exit 1
+fi
+
+# The actual Arch Linux setup starts here
+echo "[+] Installing base Arch Linux"
```
Comment (Zopieux) [inline install_scripts/bootstrap_arch_linux.sh]: This looks like you wanted to use `echo_status` for all the `[+] foo` lines, don't you?

File: install_scripts/container_setup_gw.sh [hunk 5]
```diff
@@ -0,0 +1,607 @@
...
+  echo '[-] Enable and start the mdbdns service'
+  container_run /usr/bin/systemctl enable --now mdbdns
+
+  container_run /var/prologin/venv/bin/python /var/prologin/mdb/manage.py \
+    addmachine --hostname gw --mac 11:22:33:44:55:66 \
+      --ip 192.168.1.254 --rfs 0 --hfs 0 --mtype service --room pasteur \
+      --aliases mdb,mdbsync,ns,netboot,udb,udbsync,presencesync,ntp
+
+  # Delay for the generated files to be written
+  sleep .5
```
Comment (Zopieux) [inline install_scripts/container_setup_gw.sh]: Can't we have an active loop to check these file actually exists? This kind of sleeps smells bad.

File: install_scripts/container_setup_gw.sh [hunk 6]
```diff
@@ -0,0 +1,607 @@
...
+
+  echo '[-] Enable and start the named (sic) service'
+  container_run /usr/bin/systemctl enable --now named
+
+  # Delay for named to get its little brain up and running
+  sleep .5
```
Comment (Zopieux) [inline install_scripts/container_setup_gw.sh]: Can't we have an active loop to check named is up? This kind of sleeps smells bad.

File: install.py [hunk 7]
```diff
@@ -338,6 +329,25 @@ def install_libprologin():
...
+    if not os.path.exists(SECRET_PATH):
+
+        print('We need to set the Prologin SADM master secret.\n'
+              'This secret has to be shared across all the machines.')
+        # The master secret can be configured from the environment or from user input
+        secret = os.environ.get('PROLOGIN_SADM_MASTER_SECRET', None)
+        if secret is None:
+            secret = getpass.getpass('Enter ProloginSADM master secret: ')
+            secret_check = getpass.getpass('Enter ProloginSADM master secret again: ')
+            if secret != secret_check:
```
Comment (Zopieux) [inline install.py]: Yeah good point. User can always ^C to force an error.

Comment (seirl) [inline install.py]: This code hasn't been really modified, just moved to a different place. Therefore, these comments are unrelated to the change in this PR.

File: install.py (hunk 2 above)
Comment (audebert) [inline install.py]: But where? presenced runs on the user machines and afaik there is not need of a nginx server there.

File: install_scripts/bootstrap_arch_linux.sh (hunk 4 above)
Comment (audebert) [inline install_scripts/bootstrap_arch_linux.sh]: Good catch
